pkg/manager/runnable_group_test.go

```diff
@@ -224,3 +348,64 @@ var _ = Describe("runnableGroup", func() {
 		}
 	})
 })
+
+var _ LeaderElectionRunnable = &leaderElectionRunnableFunc{}
+
+// leaderElectionRunnableFunc is a helper struct that implements LeaderElectionRunnable
+// for testing purposes.
+type leaderElectionRunnableFunc struct {
+	RunFunc                func(context.Context) error
+	NeedLeaderElectionFunc func() bool
+}
+
+func (r leaderElectionRunnableFunc) Start(ctx context.Context) error {
+	return r.RunFunc(ctx)
+}
+
+func (r leaderElectionRunnableFunc) NeedLeaderElection() bool {
+	return r.NeedLeaderElectionFunc()
+}
+
+var _ WarmupRunnable = &warmupRunnableFunc{}
+
+// warmupRunnableFunc is a helper struct that implements WarmupRunnable
+// for testing purposes.
+type warmupRunnableFunc struct {
+	RunFunc    func(context.Context) error
+	WarmupFunc func(context.Context) error
+}
+
+func (r warmupRunnableFunc) Start(ctx context.Context) error {
+	return r.RunFunc(ctx)
+}
+
+func (r warmupRunnableFunc) Warmup(ctx context.Context) error {
+	return r.WarmupFunc(ctx)
+}
+
+func (r warmupRunnableFunc) WaitForWarmupComplete(ctx context.Context) bool {
+	return true
+}
+
+// combinedRunnable implements both WarmupRunnable and LeaderElectionRunnable
```

Comment:
I would prefer leaderElectionAndWarmupRunnable

Reply:
Addressed in https://github.com/kubernetes-sigs/controller-runtime/pull/3192/commits/6bb4616ba3e651c98015a5382549982074f563ac